test/hl/test_vlentype.py

```diff
@@ -321,14 +319,88 @@ def test_variable_len_str_dset(self):
 
         f.close()
 
+    def test_variable_len_float_dset(self):
+        filename = self.getFileName("variable_len_float_dset")
+        print("filename:", filename)
+
+        f = h5py.File(filename, "w")
+
+        dims = (2,)
+        dtvlen = h5py.special_dtype(vlen=float)
+        dset = f.create_dataset('variable_len_float_dset', dims, dtype=dtvlen)
+
+        self.assertEqual(dset.name, "/variable_len_float_dset")
+        self.assertTrue(isinstance(dset.shape, tuple))
+        self.assertEqual(len(dset.shape), 1)
+        self.assertEqual(dset.shape[0], 2)
+        self.assertEqual(str(dset.dtype), 'object')
+        self.assertTrue(isinstance(dset.maxshape, tuple))
+        self.assertEqual(len(dset.maxshape), 1)
+        self.assertEqual(dset.maxshape[0], 2)
+        self.assertFalse(dset.fillvalue)  # will  be 0 for HSDS, None for h5py
+        ret_val = dset[...]
+        self.assertTrue(isinstance(ret_val, np.ndarray))
+        self.assertEqual(len(ret_val), 2)
+        e0 = ret_val[0]
+        self.assertTrue(isinstance(e0, np.ndarray))
+        self.assertEqual(e0.shape, (0,))
+
+        e0 = np.array([1.1, 2.2, 3.3], dtype=np.float64)
+        e1 = np.array([1.9, 2.8, 3.7], dtype=np.float64)
+
+        data = np.array([e0, e1], dtype=dtvlen)
+        try:
+            # This will fail on HSDS because data is a ndarray of shape (2,3) of floats
```

Comment:
Is this something that should fail due to how the API is defined, or just a bug?

Reply:
I feel like this is a bug in h5py.  I'll submit an issue, but I expect they'll tell me the answer is to just set the dataset element by element in this case.

Reply:
See this example: https://gist.github.com/jreadey/884b59a30090b0f8122c3ab51291aa6c.

`dset[...] = arr`

Attempts to write to a shape (2,) dataset with a shape (2,) numpy array, but fails with a broadcast error.
Seems like a bug in h5py, so the test case has a special handler for the test running in h5py for now.